Add Github token check for TestPrintUpStreamVersions

Currently, we are testing upstream version list directly putting HTTP request to GitHub. The more I looked to GitHub tests available currently to our repo, this looks like we need to put GitHub token check before making any GitHub API call for CI.

This should fix as the issue we are seeing in CI unit tests.

```
ok  	github.com/minishift/minishift/pkg/minishift/openshift	0.016s
--- FAIL: TestPrintUpStreamVersions (0.08s)
	Error Trace:	openshift_versions_test.go:42
	Error:      	Received unexpected error:
	            	*json.UnmarshalTypeError
	            	json: cannot unmarshal object into Go value of type []version.releaseTags
	            	&json.UnmarshalTypeError{Value:"object", Type:(*reflect.rtype)(0x7ece00), Offset:1, Struct:"", Field:""}
	Test:       	TestPrintUpStreamVersions
	Error Trace:	openshift_versions_test.go:49
	Error:      	"" does not contain "v1.4.1"
	Test:       	TestPrintUpStreamVersions
	Error Trace:	openshift_versions_test.go:50
	Error:      	"" does not contain "v3.7.2"
	Test:       	TestPrintUpStreamVersions
FAIL	github.com/minishift/minishift/pkg/minishift/openshift/version	0.396s
```

The code might be faulty, as the unmarshalling should be done in an obj,
err := unmarshall() I guess and our own error should be returned.

On Tue, Mar 27, 2018 at 4:34 PM, Lalatendu Mohanty <notifications@github.com
> wrote:

> This should fix as the issue we are seeing in CI unit tests.
>
> ok  	github.com/minishift/minishift/pkg/minishift/openshift	0.016s
> --- FAIL: TestPrintUpStreamVersions (0.08s)
> 	Error Trace:	openshift_versions_test.go:42
> 	Error:      	Received unexpected error:
> 	            	*json.UnmarshalTypeError
> 	            	json: cannot unmarshal object into Go value of type []version.releaseTags
> 	            	&json.UnmarshalTypeError{Value:"object", Type:(*reflect.rtype)(0x7ece00), Offset:1, Struct:"", Field:""}
> 	Test:       	TestPrintUpStreamVersions
> 	Error Trace:	openshift_versions_test.go:49
> 	Error:      	"" does not contain "v1.4.1"
> 	Test:       	TestPrintUpStreamVersions
> 	Error Trace:	openshift_versions_test.go:50
> 	Error:      	"" does not contain "v3.7.2"
> 	Test:       	TestPrintUpStreamVersions
> FAIL	github.com/minishift/minishift/pkg/minishift/openshift/version	0.396s
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/minishift/minishift/issues/2183#issuecomment-376442091>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAHZuPGB5kFHRgh-h1OzwoIfZvC1pOwks5tifmZgaJpZM4S8Rha>
> .
>



-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

--- a/Makefile
+++ b/Makefile
@@ -197,7 +197,7 @@ clean:
 
 .PHONY: test
 test: vendor $(ADDON_ASSET_FILE)  ## Run unit tests
-	@go test -tags "$(BUILD_TAGS)" -ldflags="$(VERSION_VARIABLES)" $(shell $(PACKAGES))
+	@go test -v -tags "$(BUILD_TAGS)" -ldflags="$(VERSION_VARIABLES)" $(shell $(PACKAGES))
 
 .PHONY: integration
 integration: GODOG_OPTS = --tags=basic ## Run integration tests

--- a/pkg/minishift/openshift/version/openshift_versions_test.go
+++ b/pkg/minishift/openshift/version/openshift_versions_test.go
@@ -23,11 +23,13 @@ import (
 	"testing"
 
 	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/util/github"
 	"github.com/minishift/minishift/pkg/version"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestPrintUpStreamVersions(t *testing.T) {
+	EnsureGitHubApiAccessTokenSet(t)
 	testDir, err := ioutil.TempDir("", "minishift-config-")
 	assert.NoError(t, err)
 	defer os.RemoveAll(testDir)
@@ -48,6 +50,13 @@ func TestPrintUpStreamVersions(t *testing.T) {
 	actualStdout := string(data)
 	assert.Contains(t, actualStdout, constants.MinimumSupportedOpenShiftVersion)
 	assert.Contains(t, actualStdout, defaultVersion)
+}
+
+func EnsureGitHubApiAccessTokenSet(t *testing.T) {
+	if github.GetGitHubApiToken() == "" {
+		t.Skip("Skipping GitHub API based test, because no access token is defined in the environment.\n " +
+			"To run this test check https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/ and set for example MINISHIFT_GITHUB_API_TOKEN (see github.go).")
+	}
 }
 
 func TestPrintDownStreamVersions(t *testing.T) {

